lib/ansible/module_utils/common/parameters.py

```diff
@@ -724,23 +724,24 @@ def _validate_sub_spec(argument_spec, parameters, prefix='', options_context=Non
             options_context.append(param)
 
             # Make sure we can iterate over the elements
-            if isinstance(parameters[param], dict):
+            if not isinstance(parameters[param], list):
```

Comment:
we should not be using dict nor list as it is easy to sneak in comparable things like set() or tuple(), Mapping and Sequence (sans strings) should be used instead

Reply:
Fixed, thanks